lua/lspconfig/server_configurations/omnisharp.lua

```diff
@@ -2,47 +2,134 @@ local util = require 'lspconfig.util'
 
 return {
   default_config = {
+    -- Enables support for reading code style, naming convention and analyzer
+    -- settings from .editorconfig.
+    enable_editorconfig_support = true,
+
+    -- If true, MSBuild project system will only load projects for files that
+    -- were opened in the editor. This setting is useful for big C# codebases
+    -- and allows for faster initialization of code navigation features only
+    -- for projects that are relevant to code that is being edited. With this
+    -- setting enabled OmniSharp may load fewer projects and may thus display
+    -- incomplete reference lists for symbols.
+    enable_ms_build_load_projects_on_demand = false,
+
+    -- Enables support for roslyn analyzers, code fixes and rulesets.
+    enable_roslyn_analyzers = false,
+
+    -- Specifies whether 'using' directives should be grouped and sorted during
+    -- document formatting.
+    organize_imports_on_format = false,
+
+    -- Enables support for showing unimported types and unimported extension
+    -- methods in completion lists. When committed, the appropriate using
+    -- directive will be added at the top of the current file. This option can
+    -- have a negative impact on initial completion responsiveness,
+    -- particularly for the first few completion sessions after opening a
+    -- solution.
+    enable_import_completion = false,
+
+    -- Specifies whether to include preview versions of the .NET SDK when
+    -- determining which version to use for project loading.
+    sdk_include_prereleases = true,
+
+    -- Only run analyzers against open files when 'enableRoslynAnalyzers' is
+    -- true
+    analyze_open_documents_only = false,
+
     filetypes = { 'cs', 'vb' },
     root_dir = function(fname)
       return util.root_pattern '*.sln'(fname) or util.root_pattern '*.csproj'(fname)
     end,
     on_new_config = function(new_config, new_root_dir)
-      if new_root_dir then
-        table.insert(new_config.cmd, '-s')
-        table.insert(new_config.cmd, new_root_dir)
+      table.insert(new_config.cmd, '-z') -- https://github.com/OmniSharp/omnisharp-vscode/pull/4300
+      vim.list_extend(new_config.cmd, { '-s', new_root_dir })
+      vim.list_extend(new_config.cmd, { '--hostPID', tostring(vim.fn.getpid()) })
+      table.insert(new_config.cmd, 'DotNet:enablePackageRestore=false')
```

Comment:
@williamboman What is the reason for package restore to be disabled by default? I know it has had a turbulent history, is it still not mature enough for usage?

Alternatively, it could be turned into a config property like the others you added, but kept off by default, and possibly with a warning description.

Reply:
> @williamboman What is the reason for package restore to be disabled by default? I know it has had a turbulent history, is it still not mature enough for usage?

It's taken from how the VSCode extension launches the server, no other reason than that.

> Alternatively, it could be turned into a config property like the others you added, but kept off by default, and possibly with a warning description.

Sounds reasonable to me, I'm not a maintainer though. I'm sure if you open a PR it'd get merged

Reply:
Thanks for clarifying, will try it out and see if it's worth it.